### ISSUE TITLE
[FEATURE] Support for wait_for_completion flag in force merge API

### ISSUE THREAD

**Issue by navneet1v** [opensearch-project/opensearch-py#509]
### Is your feature request related to a problem?
Currently the force merge api in the client waits for force_merge to get completed. But OpenSearch provides a way where if we pass wait_for_completion = true, it gives a task id and poll the task id to see if the force merge is completed or not.

### What solution would you like?
Add the support for wait_for_completion flag in force merge api.

```
POST /hotels-index/_forcemerge?max_num_segments=1&wait_for_completion=true
```

### What alternatives have you considered?
The only other way is to send Raw HTTP request, but the domain is backed with security creating raw HTTP request is very difficult and not feasible.

### Do you have any additional context?
NA

**Comment by dblock** [comment]
@saimedhi What's the right way to bring this in with the new generated code?

**Comment by saimedhi** [comment]
@dblock To add 'wait_for_completion' parameter to the 'forcemerge' API in Python, one approach is through the Python Generator. For a more comprehensive solution, it's best to integrate this parameter into the OpenSearch OpenAPI spec, making it accessible across all language client libraries.

**Comment by dblock** [comment]
@saimedhi yes exactly, is there a walkthrough / dev guide of what to add where?

**Comment by nhtruong** [comment]
Yes we have a developer_guide on the api repo: https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md

**Comment by saimedhi** [comment]
@dblock, @nhtruong please transfer this issue to [opensearch-api-specification](https://github.com/opensearch-project/opensearch-api-specification/tree/main). I don't have access to do it.

**Comment by navneet1v** [comment]
@saimedhi once this issue is moved there how long it will take for the attribute to be available in python client?

**Comment by saimedhi** [comment]
Hello @navneet1v,

- Once a PR is merged into the spec, we can generate and update the Python client whenever needed.
- If we directly add changes to the Python client, they may get overwritten when the generator runs. So, it's best to make the initial change in the spec.

If you're familiar with Smithy, consider contributing. This way, we can ensure our PR gets merged quickly.
